examples/8a_benchmark_write1D2D_parallel.cpp

```diff
@@ -0,0 +1,682 @@
+  {
+    auto E = std::shared_ptr< T > {
+      new T[size], []( T * d ) {delete[] d;}
+    };
+
+    for(unsigned long  i = 0ul; i < size; i++ )
+      {
+        E.get()[i] = val;
+      }
+    return E;
+  }
+
+/** Find supported backends
+ *  (looking for ADIOS2 or H5)
+ *
+ */
+std::vector<std::string> getBackends() {
+    std::vector<std::string> res;
+#if openPMD_HAVE_ADIOS2
+    if( auxiliary::getEnvString( "OPENPMD_BP_BACKEND", "NOT_SET" ) != "ADIOS1" )
+        res.emplace_back(".bp");
+#endif
+
+#if openPMD_HAVE_HDF5
+    res.emplace_back(".h5");
+#endif
+    return res;
+}
+
+/** ... Class TestInput
+ *
+ *
+ * @param mpi_size .... MPI size
+ * @param mpi_rank .... MPI rank
+ * @param bulk ........ num of elements
+ * @param numSeg ...... num of subdivition for the elements
+ *               ...... note that with h5collect mode, numSeg must be 1
+ * @param numSteps .... num of iterations
+ */
+class TestInput
+{
+public:
+  TestInput() =  default;
+
+  unsigned int GetSeg() const
+  {
+    if (m_Backend == ".h5")
+        if (auxiliary::getEnvString( "OPENPMD_HDF5_INDEPENDENT", "ON" ) != "ON")
+            return 1;
+    return m_Seg;
+  }
+
+  int m_MPISize = 1;
+  int m_MPIRank = 0;
+  unsigned long m_Bulk = 1000ul;
+  unsigned int m_Seg = 1;
+  int m_Steps = 1;
+  int m_TestNum = 0;
+  std::string m_Backend = ".bp";
+};
+
+/** divide "top" elements into "upTo" non-zero segments
+ *
+ *
+ *
+ * @param top ...  number of elements to be subdivided
+ * @param upTo ... subdivide into this many different blocks
+ * @param repeats ... roll the die this many times to avoid duplicates between ranks
+ * @return ... returns the vector that has subdivision information
+ */
+std::vector< unsigned long >
+segments( unsigned long top, unsigned int upTo, int& repeats )
+{
+    std::vector< unsigned long > result;
+
+    if( upTo == 0 || top < upTo )
+        return result;
+
+    // how many partitions
+    std::default_random_engine generator;
+    std::uniform_int_distribution< int > distribution( 1, upTo );
+    auto dice = std::bind ( distribution, generator );
+
+    int howMany = dice();
+    // repeat to avoid duplicates btw ranks
+    for( auto i=0; i<repeats; i++ )
+        howMany = dice();
+
+    if( howMany == 0 )
+        howMany = 1;
+
+    if( howMany == 1 ) {
+        result.push_back(top);
+        return result;
+    }
+
+    unsigned long counter = 0ul;
+
+    for( int i=0; i<howMany; i++ ) {
+        if( counter < top ) {
+            if( i == howMany - 1 )
+                result.push_back(top - counter);
+            else {
+                auto curr = rand() % (top-counter);
+                result.push_back(curr);
+                counter += curr;
+            }
+        } else
+            result.push_back( 0u );
+    }
+
+    return result;
+}
+
+
+/** Load data into series
+ *
+ * all tests call this functions to store and flush 1D data
+ *
+ * @param series ..... opemPMD-api series
```

Comment:
please just use spaces instead of `...` (you can align the spaces if you like, but not needed)
the syntax here will be parsed and shown at https://openpmd-api.readthedocs.io/en/0.11.1-alpha/_static/doxyhtml/annotated.html (linked in [manual](https://openpmd-api.readthedocs.io/en/0.11.1-alpha/details/doxygen.html))